room.go

```diff
@@ -43,14 +44,20 @@ func WithAutoSubscribe(val bool) ConnectOption {
 	}
 }
 
+func WithCallback(cb *RoomCallback) ConnectOption {
+	return func(p *ConnectParams) {
+		p.Callback = cb
+	}
+}
+
 type PLIWriter func(webrtc.SSRC)
 
 type Room struct {
 	engine           *RTCEngine
-	SID              string
-	Name             string
+	sid              string
+	name             string
 	LocalParticipant *LocalParticipant
-	Callback         *RoomCallback
+	callback         *RoomCallback
```

Comment:
This will break existing applications. We need to update documentation showing them how to use this, especially using `NewRoomCallback()` and overriding fields before passing it in.